lightning/src/lib.rs

```diff
@@ -28,31 +28,105 @@
 #![allow(bare_trait_objects)]
 #![allow(ellipsis_inclusive_range_patterns)]
 
+#![cfg_attr(all(not(feature = "std"), not(test)), no_std)]
```

Comment:
We should have an explicit error for `not(feature = "std") && not(feature = "no-std")`

Reply:
added